### ISSUE TITLE
Data Management v2

### ISSUE THREAD

**Issue by guillaume-byte** [GrayboxTech/weightslab#50]
SDK data is two things: 
  1- Dataloaders from datasets, i.e., yield image, labels, uid, pre/post process
  2- Data storage, i.e., storing training data like loss per image for instance.

The problem is that 2- is a dependent and overlayer of 1-, i.e., data storage is split into train, test, and val. We should have a ledger dataframe with every uid@data.
